File: api/quizzler/src/main/java/com/quizzler/api/domain/QuizSession.java
package com.quizzler.api.domain;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "quiz_session")
public class QuizSession {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "public_id", nullable = false, unique = true)
    private String publicId;

    protected QuizSession() {
    }

    public QuizSession(String publicId) {
        this.publicId = publicId;
    }

    public Long getId() {
        return id;
    }

    public String getPublicId() {
        return publicId;
    }
}

File: ui/quizzler/src/app/entities/quizsession.ts
export class QuizSession {
    constructor(
        public publicId: string
    ) { };

    public static getDefaultQuizSession() : QuizSession {
            return new QuizSession('0');
    }

    public isDefault() : boolean {
        return this.publicId === QuizSession.getDefaultQuizSession().publicId;
    }
}
